client/src/components/GuestMenu.jsx

```diff
@@ -16,6 +16,14 @@ const GuestMenu = ({ guests, joinName, menu, sessionId }) => {
   const [currentOrder, setCurrentOrder] = useState([]);
   const [submitted, setSubmitted] = useState(false);
 
+  useEffect(() => {
+    socket.emit('joinRoom', { sessionId });
+  });
```

Comment:
Without any dependencies, this will run after *anything* in the DOM renders. YOu can add an empty array, and it is effectively a `ComponentDidMount`(see [this](https://stackoverflow.com/questions/58579426/in-useeffect-whats-the-difference-between-providing-no-dependency-array-and-an))
```suggestion
  useEffect(() => {
    socket.emit('joinRoom', { sessionId });
  }, []);
```

Reply:
Added